hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1586,6 +1577,25 @@ private static Stream<String> getFirstNFieldNames(Stream<String> fieldNames, int
     return fieldNames.filter(fieldName -> !HOODIE_META_COLUMNS_WITH_OPERATION.contains(fieldName)).limit(n);
   }
 
+  @VisibleForTesting
+  public static Pair<String, Schema.Field> getSchemaForField(Schema schema, String fieldName) {
+    return getSchemaForField(schema, fieldName, StringUtils.EMPTY_STRING);
+  }
+
+  @VisibleForTesting
+  public static Pair<String, Schema.Field> getSchemaForField(Schema schema, String fieldName, String prefix) {
```

Comment:
Let's move these util methods to `HoodieAvroUtils` in hudi-common. And, can we add some direct UTs for the methods with simple field, nested field, nullable/non-nullable.